### PR TITLE
ymdk/bface - RGB config and minila layout

### PR DESCRIPTION
**Pull request by shohart**
## Description

Just some RGB fix as @fauxpark helped to implement, and added minila layout.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ymdk/bface/keymaps/minila/keymap.c [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include QMK_KEYBOARD_H
+
+
+#define _FL 1
+#define  ______ KC_TRNS
```
Comment (zvecr) [inline keyboards/ymdk/bface/keymaps/minila/keymap.c:22]:
```suggestion
enum layer_names {
  _BL,
  _FL
};
```

File: keyboards/ymdk/bface/keymaps/minila/keymap.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    //Base Layer
+    [0] = LAYOUT_all(
+		KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_GRV, KC_BSPC, 
+		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, 
+		LT(1, KC_CAPS), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NO, MT(MOD_RCTL, KC_ENT), 
+		KC_LSPO, KC_LSPO, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSPC, KC_UP, KC_DEL, 
+		KC_LCTL, KC_LGUI, KC_LALT, MO(1), KC_SPC, MO(1), KC_RALT, KC_APP, KC_LEFT, KC_DOWN, KC_RGHT
+    ),
```
Comment (zvecr) [inline keyboards/ymdk/bface/keymaps/minila/keymap.c:32]:
```suggestion
    [_BL] = LAYOUT_all(
		KC_ESC, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL, KC_GRV, KC_BSPC, 
		KC_TAB, KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS, 
		LT(_FL, KC_CAPS), KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT, KC_NO, MT(MOD_RCTL, KC_ENT), 
		KC_LSPO, KC_LSPO, KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH, KC_RSPC, KC_UP, KC_DEL, 
		KC_LCTL, KC_LGUI, KC_LALT, MO(_FL), KC_SPC, MO(_FL), KC_RALT, KC_APP, KC_LEFT, KC_DOWN, KC_RGHT
    ),
```

File: keyboards/ymdk/bface/keymaps/minila/keymap.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    //FN Layer
+    [_FL] = LAYOUT_all(
+		KC_TILD, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, ______, RESET, 
+		______, ______, KC_HOME, KC_UP, KC_END, KC_PGUP, ______, NK_TOGG, KC_PSCR, KC_SLCK, KC_PAUS, ______, BL_STEP, BL_TOGG, 
+		______, ______, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, ______, KC_INS, KC_HOME, KC_PGUP, KC_BSPC, ______, ______, ______, 
+		______, ______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAD, RGB_VAI, ______, KC_DEL, KC_END, KC_PGDN, ______, ______, ______, ______, 
+		______, ______, ______, ______, KC_ENT, ______, ______, ______, ______, ______, ______
+    ),
```
Comment (zvecr) [inline keyboards/ymdk/bface/keymaps/minila/keymap.c:40]:
```suggestion
    [_FL] = LAYOUT_all(
		KC_TILD, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, _______, RESET, 
		_______, _______, KC_HOME, KC_UP, KC_END, KC_PGUP, _______, NK_TOGG, KC_PSCR, KC_SLCK, KC_PAUS, _______, BL_STEP, BL_TOGG, 
		_______, _______, KC_LEFT, KC_DOWN, KC_RGHT, KC_PGDN, _______, KC_INS, KC_HOME, KC_PGUP, KC_BSPC, _______, _______, _______, 
		_______, _______, RGB_TOG, RGB_MOD, RGB_HUI, RGB_VAD, RGB_VAI, _______, KC_DEL, KC_END, KC_PGDN, _______, _______, _______, _______, 
		_______, _______, _______, _______, KC_ENT, _______, _______, _______, _______, _______, _______
    ),
```

**Comment by fauxpark** [comment]
I can't see any unpopulated resistor footprints on the PCB in the readme image, so it *might* be okay. But it's too low res to make out the values.

**Comment by shohart** [comment]
> I can't see any unpopulated resistor footprints on the PCB in the readme image, so it _might_ be okay. But it's too low res to make out the values.

Honestly, i never saw this PCB without RGB LEDs... YMDK offers only RGB versions in their kits.

File: keyboards/ymdk/bface/keymaps/minila/readme.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+Basically layout like on Filco Minila board, found here: https://geekboards.ru/product/filco-minila
+There is one single function layer that can be accessed by pressing both FN keys and also by dolding down CAPS LOCK. 
+Enter is aslo a Right contol when held down. Shifts are also '(' and ')' respectevly when tapped.
```
Comment (noroadsleft) [inline keyboards/ymdk/bface/keymaps/minila/readme.md:3]:
```suggestion
Basically layout like on Filco Minila board, found here: https://geekboards.ru/product/filco-minila

There is one single function layer that can be accessed by holding either FN key or Caps Lock. Enter is also a Right Control when held down. Shifts are also '(' and ')' respectively when tapped.

```

File: keyboards/ymdk/bface/keymaps/minila/readme.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+You can control underglow RGBs:
+FN+Z - Toggle
+FN+X - Effect
+FN+C - Color
+FN+V - Brightness lower
+FN+B - Brightness higher
```
Comment (noroadsleft) [inline keyboards/ymdk/bface/keymaps/minila/readme.md:9]:
```suggestion

* FN+Z - Toggle
* FN+X - Effect
* FN+C - Color
* FN+V - Brightness lower
* FN+B - Brightness higher
```

**Comment by shohart** [comment]
> Approved, providing we react quickly to any users complaining that their "no led" board has stopped functioning.

Yeah, ok, ugh, maybe i should make a bface_norgb keabord under ymdk/ then?

**Comment by fauxpark** [comment]
> Yeah, ok, ugh, maybe i should make a bface_norgb keabord under ymdk/ then?

Only if we get complaints.

> BTW, is NKRO for ps2avrGB boards is a considared thing? because now FW can not be compiled because of protocol issue...

For the most part, it just hasn't been implemented yet. V-USB boards also have more of a limitation on the number of endpoints they can have, due to the USB being implemented in software rather than hardware as with USB AVRs or ARM chips. But this is being worked on.

**Comment by shohart** [comment]
Okay, i got it.

**Comment by mechmerlin** [comment]
oof is this board really called the bface? In before lots of people confusing their wkl bface for this one.

**Comment by zvecr** [comment]
Thanks!

**Pull request merged**